pkg/profiler/profiler.go

```diff
@@ -261,6 +261,11 @@ func (p *CgroupProfiler) Run(ctx context.Context) error {
 		if err != nil {
 			return fmt.Errorf("open perf event: %w", err)
 		}
+		defer func() {
```

Comment:
Sorry, just noticed something here! If I am not mistaken, this might not be what we want, and we won't close all the FDs we want! What is worse, we might be accidentally closing FDs we don't want to (if we race with another process).

I _think_ we want to pass the FD as an argument, such as: https://go.dev/play/p/tB-jjUUaNZ4 instead of https://go.dev/play/p/e9vpgsF3364.

This is due to the way defer seems to work, as it evaluates the arguments at the time `defer` is assembled, instead of when the function is finally called, but currently we used the variable that's in the scope, which will always be the last FD.

I might be 100% wrong, let's learn some Go together 😄

Reply:
:see_no_evil: Amazing catch! I missed that we're in a loop!

<img src="https://media0.giphy.com/media/80TEu4wOBdPLG/giphy.gif"/>

Reply:
Now that, I've checked it again. We use `fd` in the defer closure and `fd` is local so, it should be a problem.

Reply:
https://go.dev/play/p/VxHz5xAekZH

Reply:
You are 100% right! https://go.dev/play/p/ucsC6MeLYsX My bad :)